processor/resourcedetectionprocessor/internal/system/metadata_others.go

```diff
@@ -0,0 +1,35 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package system
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func (*systemMetadataImpl) FQDNAvailable() bool {
+	_, err := os.Stat("/bin/hostname")
```

Comment:
This wouldn't work in our Docker images because we don't include a base OS with utilities like hostname. But we also don't include /etc/hosts, etc. so I'm not sure we'd have a fqdn anyway. 🤔 

However maybe we should use https://github.com/Showmax/go-fqdn since it has Windows support (as well as not relying on hostname binary).

@keitwb any ideas?

Reply:
> This wouldn't work in our Docker images because we don't include a base OS with utilities like hostname. But we also don't include /etc/hosts, etc. so I'm not sure we'd have a fqdn anyway. 🤔

I am aware of this  (you would have to install the `net-tools` package on the image) but I was thinking of other use cases (e.g. users that rely on `.deb` or `.rpm` packages). The project does plan to support the latter, right?

> However maybe we should use https://github.com/Showmax/go-fqdn since it has Windows support (as well as not relying on hostname binary).

That sounds like a good idea to me.

Reply:
Yep we publish deb and rpm so it would be available in some cases. But ideally we would want the processor to behave the same across as many environments as possible.

Reply:
👍 I am going to use `go-fqdn` then if @keitwb does not have any other ideas, which would have more widespread support and seems reasonable